### DISCUSSION TITLE
LDAP Encryption in config.json

### DISCUSSION THREAD

**Discussion by diegofreesz** [semaphoreui/semaphore#1390] [General]
Hi there,

in config.json "ldap_bindpassword": Is it possible to provide an encrypted password?

if i provide a password in clear text, any sniffer will be able to read it.

**Comment by ansibleguy** [comment]
When using LDAP you should always use LDAPS (encrypted) so the passwords and user-information are protected in transit.
LDAPS can be enabled using the config => see ['ldap_needtls'](https://docs.ansible-semaphore.com/administration-guide/configuration#configuration-file)

I don't think you can provide an encrypted password. (_at rest encryption_)
Can you give a practical example how someone can 'sniff' the content of a file, that is only readable by root & the service-user running semaphore?

**Comment by tboerger** [comment]
It is not possible to encrypt this password as it needs to be provided in plaintext to the LDAP client used within Semaphore.
